encoder.go

```diff
@@ -50,3 +60,47 @@ func (e *Encoder) writeLiteralFieldWithoutNameReference(f HeaderField) {
 	e.buf = appendVarInt(e.buf, 7, uint64(len(f.Value)))
 	e.buf = append(e.buf, []byte(f.Value)...)
 }
+
+// Encodes a header field whose name is present in one of the
+// tables. TBit is true if the idx refers to the static table.
+func (e *Encoder) writeLiteralFieldWithNameReference(
+	f *HeaderField, idx int, NBit, TBit bool) {
+	offset := len(e.buf)
+	e.buf = appendVarInt(e.buf, 4, uint64(idx))
+	// Set the 01NTxxxx pattern
+	e.buf[offset] ^= 0x40
+	if TBit {
+		e.buf[offset] ^= 0x10
+	}
+	if NBit {
+		e.buf[offset] ^= 0x20
+	}
+	e.buf = appendVarInt(e.buf, 7, uint64(len(f.Value)))
+	e.buf = append(e.buf, []byte(f.Value)...)
+}
+
+// Encodes an indexed field, meaning it's entirely defined in one of the
+// tables. TBit is true if the idx refers to the static table.
+func (e *Encoder) writeIndexedField(idx int, TBit bool) {
```

Comment:
It seems like you don’t need the TBit parameter here, as it’s always set to true.